pkg/wal/processor/search/store/search_store.go

```diff
@@ -9,63 +9,75 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/xataio/pgstream/internal/es"
+	"github.com/xataio/pgstream/internal/searchstore"
+	elasticsearchstore "github.com/xataio/pgstream/internal/searchstore/elasticsearch"
+	opensearchstore "github.com/xataio/pgstream/internal/searchstore/opensearch"
+
 	loglib "github.com/xataio/pgstream/pkg/log"
 	"github.com/xataio/pgstream/pkg/schemalog"
 	"github.com/xataio/pgstream/pkg/wal/processor/search"
 )
 
 type Store struct {
-	logger           loglib.Logger
-	client           es.SearchClient
-	mapper           search.Mapper
-	adapter          SearchAdapter
-	indexNameAdapter IndexNameAdapter
-	marshaler        func(any) ([]byte, error)
+	logger               loglib.Logger
+	client               searchstore.Client
+	mapper               search.Mapper
+	adapter              SearchAdapter
+	indexNameAdapter     IndexNameAdapter
+	marshaler            func(any) ([]byte, error)
+	defaultIndexSettings map[string]any
 }
 
 type Config struct {
-	URL string
+	OpenSearchURL    string
+	ElasticsearchURL string
 }
 
 type Option func(*Store)
 
 const (
-	openSearchDefaultANNEngine      = "nmslib"
-	openSearchDefaultM              = 48
-	openSearchDefaultEFConstruction = 256
-	openSearchDefaultEFSearch       = 100
-
-	// OpenSearch has a limit of 512 bytes for the ID field. see here:
+	// OpenSearch/Elasticsearch have a limit of 512 bytes for the ID field. see here:
 	// https://www.elastic.co/guide/en/elasticsearch/reference/7.10/mapping-id-field.html
-	osIDFieldLengthLimit = 512
+	idFieldLengthLimit = 512
 
 	schemalogIndexName = "pgstream"
 )
 
 func NewStore(cfg Config, opts ...Option) (*Store, error) {
-	os, err := es.NewClient(cfg.URL)
+	var searchStore searchstore.Client
+	var err error
+	switch {
+	case cfg.OpenSearchURL != "":
+		searchStore, err = opensearchstore.NewClient(cfg.OpenSearchURL)
+	case cfg.ElasticsearchURL != "":
+		searchStore, err = elasticsearchstore.NewClient(cfg.ElasticsearchURL)
+	default:
+		return nil, errors.New("invalid search store configuration provided")
```

Comment:
it would probably be good to provide more detail in all the different error cases here:

* Both OpenSearch + Elasticsearch are configured ❗ 
* None of them is ❗